src/dbnode/persist/fs/wide/index_checksum_block_buffer_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+
+package wide
+
+import (
+	"testing"
+
+	"github.com/m3db/m3/src/x/ident"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIndexChecksumBlockBuffer(t *testing.T) {
+	ch := make(chan ident.IndexChecksumBlock)
+	buf := NewIndexChecksumBlockBuffer(ch)
+	bl := ident.IndexChecksumBlock{Marker: []byte("foo")}
+	go func() {
+		ch <- bl
+		close(ch)
+	}()
+
+	assert.True(t, buf.Next())
+	assert.Equal(t, []byte("foo"), buf.Current().Marker)
```

Comment:
Nit: maybe just
```suggestion
	assert.Equal(t, bl, buf.Current())
```
?